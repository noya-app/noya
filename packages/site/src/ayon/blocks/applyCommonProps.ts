import { CommonProps } from '@noya-design-system/protocol';
import { BlockProps } from 'noya-state';

export function applyCommonProps(
  props: BlockProps,
): Pick<CommonProps, '_passthrough'> {
  return {
    _passthrough: {
      ...(props.dataSet && {
        key: props.dataSet.id,
        'data-noya-id': props.dataSet.id,
        'data-noya-parent-id': props.dataSet.parentId,
      }),
    },
  };
}
