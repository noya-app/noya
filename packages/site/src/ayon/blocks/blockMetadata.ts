import { Size } from 'noya-geometry';
import {
  avatarSymbolId,
  boxSymbolId,
  buttonSymbolId,
  cardSymbolId,
  checkboxSymbolId,
  featureItemSymbolId,
  headerBarSymbolId,
  heading1SymbolId,
  heading2SymbolId,
  heading3SymbolId,
  heading4SymbolId,
  heading5SymbolId,
  heading6SymbolId,
  heroSymbolId,
  heroWithImageSymbolId,
  iconSymbolId,
  imageSymbolId,
  inputSymbolId,
  linkSymbolId,
  radioSymbolId,
  selectSymbolId,
  sidebarSymbolId,
  signInSymbolId,
  switchSymbolId,
  tableSymbolId,
  tagSymbolId,
  textareaSymbolId,
  textSymbolId,
  tileCardSymbolId,
  writeSymbolId,
} from '../symbols/symbolIds';

export type BlockCategory = 'application' | 'marketing' | 'element';

export type PreferredOverride = {
  symbolId: string;
  blockText: string;
  resolvedBlockText?: string;
};

export type BlockMetadata = {
  name: string;
  category: BlockCategory;
  preferredSize: Size;
  preferredBlockText?: string;
  preferredResolvedBlockText?: string;
  preferredOverrides?: PreferredOverride[];
  hideInDocs?: boolean;
};

const headingMetadata: BlockMetadata = {
  name: 'Heading',
  category: 'element',
  preferredSize: { width: 700, height: 100 },
  preferredBlockText: 'All About Cats',
};

const headingNMetadata: BlockMetadata = {
  ...headingMetadata,
  hideInDocs: true,
};

// This is in a separate file so docs don't have to import Chakra components directly.
// Docs can import this synchronously and import(Ayon)
export const blockMetadata: Record<string, BlockMetadata> = {
  [avatarSymbolId]: {
    name: 'Avatar',
    category: 'element',
    preferredSize: { width: 64, height: 64 },
  },
  [buttonSymbolId]: {
    name: 'Button',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [linkSymbolId]: {
    name: 'Link',
    category: 'element',
    preferredSize: { width: 200, height: 40 },
  },
  [boxSymbolId]: {
    name: 'Box',
    category: 'element',
    preferredSize: { width: 200, height: 200 },
  },
  [heroSymbolId]: {
    name: 'Hero',
    category: 'marketing',
    preferredSize: { width: 600, height: 400 },
  },
  [heroWithImageSymbolId]: {
    name: 'Hero with Image',
    category: 'marketing',
    preferredSize: { width: 1280, height: 720 },
  },
  [tagSymbolId]: {
    name: 'Tag',
    category: 'element',
    preferredSize: { width: 140, height: 30 },
  },
  [cardSymbolId]: {
    name: 'Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 300 },
    preferredOverrides: [
      {
        symbolId: imageSymbolId,
        blockText: 'landscape',
        resolvedBlockText:
          'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
      },
    ],
  },
  [tileCardSymbolId]: {
    name: 'Tile Card',
    category: 'marketing',
    preferredSize: { width: 250, height: 250 },
  },
  [iconSymbolId]: {
    name: 'Icon',
    category: 'element',
    preferredSize: { width: 48, height: 48 },
  },
  [imageSymbolId]: {
    name: 'Image',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'landscape',
    preferredResolvedBlockText:
      'https://images.unsplash.com/photo-1514917860136-ee8b88e8c9c9?crop=entropy&cs=tinysrgb&fit=crop&fm=jpg&h=300&ixid=MnwxfDB8MXxyYW5kb218MHx8bGFuZHNjYXBlLGJ1c3Rlcnx8fHx8fDE2Nzc2NDIxNDM&ixlib=rb-4.0.3&q=80&utm_campaign=api-credit&utm_medium=referral&utm_source=unsplash_source&w=300',
  },
  [writeSymbolId]: {
    name: 'Write',
    category: 'element',
    preferredSize: { width: 300, height: 300 },
    preferredBlockText: 'Cats are great pets',
    preferredResolvedBlockText:
      'Cats are incredibly intelligent creatures with personalities all of their own. They make loyal and loving companions who can provide years of fun and joy. Cats come in all shapes, sizes, and colors and offer a variety of health benefits to their human owners. From their natural mouse-catching abilities to their playful antics, cats make terrific pet choices for a wide variety of households.',
  },
  [textSymbolId]: {
    name: 'Text',
    category: 'element',
    preferredSize: { width: 300, height: 150 },
    preferredBlockText:
      'Cats are great pets. They make loyal and loving companions who can provide years of fun and joy.',
  },
  [heading1SymbolId]: headingMetadata,
  [heading2SymbolId]: headingNMetadata,
  [heading3SymbolId]: headingNMetadata,
  [heading4SymbolId]: headingNMetadata,
  [heading5SymbolId]: headingNMetadata,
  [heading6SymbolId]: headingNMetadata,
  [signInSymbolId]: {
    name: 'Sign In',
    category: 'application',
    preferredSize: { width: 350, height: 250 },
  },
  [tableSymbolId]: {
    name: 'Table',
    category: 'application',
    preferredSize: { width: 600, height: 400 },
  },
  [sidebarSymbolId]: {
    name: 'Sidebar',
    category: 'application',
    preferredSize: { width: 250, height: 600 },
  },
  [headerBarSymbolId]: {
    name: 'Header Bar',
    category: 'application',
    preferredSize: { width: 800, height: 80 },
  },
  [checkboxSymbolId]: {
    name: 'Checkbox',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [radioSymbolId]: {
    name: 'Radio',
    category: 'element',
    preferredSize: { width: 300, height: 30 },
  },
  [selectSymbolId]: {
    name: 'Select',
    category: 'element',
    preferredSize: { width: 300, height: 40 },
  },
  [textareaSymbolId]: {
    name: 'Textarea',
    category: 'element',
    preferredSize: { width: 240, height: 120 },
  },
  [inputSymbolId]: {
    name: 'Input',
    category: 'element',
    preferredSize: { width: 240, height: 40 },
  },
  [switchSymbolId]: {
    name: 'Switch',
    category: 'element',
    preferredSize: { width: 50, height: 35 },
  },
  [featureItemSymbolId]: {
    name: 'Feature Item',
    category: 'marketing',
    preferredSize: { width: 400, height: 160 },
  },
};
