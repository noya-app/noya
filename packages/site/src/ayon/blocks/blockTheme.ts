// Blue
export const accentColor = {
  50: '#eff6ff',
  100: '#dbeafe',
  200: '#bfdbfe',
  300: '#93c5fd',
  400: '#60a5fa',
  500: '#3b82f6',
  600: '#2563eb',
  700: '#1d4ed8',
  800: '#1e40af',
  900: '#1e3a8a',
};

// Slate
export const neutralColor = {
  50: '#f8fafc',
  100: '#f1f5f9',
  200: '#e2e8f0',
  300: '#cbd5e1',
  400: '#94a3b8',
  500: '#64748b',
  600: '#475569',
  700: '#334155',
  800: '#1e293b',
  900: '#0f172a',
};

export const buttonColors = {
  light: {
    backgroundColor: '#f1f5f9',
    color: '#000',
  },
  lightDisabled: {
    backgroundColor: '#e2e8f0',
    color: '#000',
  },
  dark: {
    backgroundColor: '#475569',
    color: '#fff',
  },
  darkDisabled: {
    backgroundColor: '#8799b2',
    color: '#fff',
  },
};
