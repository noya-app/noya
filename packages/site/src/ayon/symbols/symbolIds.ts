import { component } from '@noya-design-system/protocol';

export const ayonLibraryId = 'e6cc1791-d9c1-410a-84b6-b381fe058bef';

export const buttonSymbolId = 'cdf2346b-cb21-4f23-8d93-7c4fb2e3a5a0';
export const avatarSymbolId = 'fe7dd31d-b140-4e4c-ab7b-90fbe833936b';
export const boxSymbolId = '9dc8f73e-64f2-43f2-8902-af334a7a17cd';
export const checkboxSymbolId = 'fa181f85-04c2-4524-ad4f-779240cb44b0';
export const inputSymbolId = '4db59e9d-9bc1-4146-af0a-ec8bf801485f';
export const switchSymbolId = '30a8ac0b-18dd-4744-bee1-77852ec4d3d8';
export const textSymbolId = 'a1f7266f-50cc-416f-9ed2-4af4bca30257';
export const imageSymbolId = 'd91ba1e3-7e64-4966-9cc1-daa48f989178';
export const heading1SymbolId = '955f4cb1-7879-4331-a359-a21a031147cb';
export const heading2SymbolId = 'b3884865-3fbe-4ed0-b577-c3f5864c4a21';
export const heading3SymbolId = 'ab7fb087-28b6-4c9a-a925-22311efb6a19';
export const heading4SymbolId = 'b8441949-12ba-42be-8f87-5cf222af70d0';
export const heading5SymbolId = 'a6f2653e-1595-4dcc-b979-4e0b1340fda3';
export const heading6SymbolId = '467ed35a-0351-4997-ab1e-44b2c6779111';
export const writeSymbolId = 'a0951cdf-6023-4cda-b5fc-8c2a8d0d9986';
export const headerBarSymbolId = 'ab3fbd14-5b81-4c58-873c-95683f4b03d7';
export const heroSymbolId = 'd15edd3e-df47-438a-9c92-059b4c14915d';
export const heroSymbolV2Id = '61eb943b-7ee7-4b33-9315-aac7cb2d2764';
export const heroWithImageSymbolId = 'ce98b3b6-b831-43c7-8c87-ea79e20f6601';
export const tagSymbolId = component.id.Tag;
export const iconSymbolId = '18682473-676a-46d3-8faf-68dc5b1c204f';
export const sidebarSymbolId = '5733e3d6-9688-4c6d-b031-6be27d05767a';
export const tableSymbolId = 'c152b560-d8af-4b16-8202-55111647d17a';
export const selectSymbolId = '7e913e19-97c6-442f-8cc4-58c4f2e71e77';
export const radioSymbolId = 'e2fbabf8-81f4-4d43-a56d-6067737df0ea';
export const textareaSymbolId = '7bcb8f7b-7645-429b-89f0-2298fef4ccb6';
export const spacerSymbolId = '15907162-13f5-4366-81af-19ced2192245';
export const cardSymbolId = '8bbffbff-efc8-4d9b-9972-73a655adaf47';
export const tileCardSymbolId = '09dcb31d-5bb2-43f2-a30a-0a9e00d92cde';
export const signInSymbolId = '78481f59-0db6-4de6-b38c-f4e4d9354fa9';
export const linkSymbolId = '5be80b87-7bd6-4090-8947-316d54064544';
export const featureItemSymbolId = 'bb9acbb5-dea6-48ac-a525-124cf9ff6054';
export const featureRowSymbolId = '128c943f-acfc-4618-98b7-ce4150273913';
export const featureSectionSymbolId = '165dc64a-569c-4dbc-a008-11944ca143d0';
