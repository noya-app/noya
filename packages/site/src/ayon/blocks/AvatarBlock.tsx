import { applyCommonProps } from '@noya-design-system/protocol';
import { BlockDefinition } from 'noya-state';
import { isExternalUrl } from 'noya-utils';
import { getBlockClassName } from '../tailwind/tailwind';
import { getParameters } from '../utils/getMappedParameters';
import { isApproximatelySquare, isWithinRectRange } from './score';
import { avatarSymbolId } from './symbolIds';
import { avatarSymbol } from './symbols';

const AVATAR_SIZES = ['2xs', 'xs', 'sm', 'md', 'lg', 'xl', '2xl'];

export const AvatarBlock: BlockDefinition = {
  symbol: avatarSymbol,
  infer: ({ frame, blockText }) =>
    isWithinRectRange({
      rect: frame,
      minWidth: 30,
      minHeight: 30,
      maxWidth: 120,
      maxHeight: 120,
    }) && isApproximatelySquare(frame, 0.2)
      ? 0.8
      : 0,
  render: ({ h, Components: { [avatarSymbolId]: Avatar } }, props) => {
    const parameters = getParameters(props.blockParameters);
    const content = props.blockText ?? '';
    const src = isExternalUrl(content) ? content : undefined;
    const name = !src ? content : undefined;
    const hashtags = Object.keys(parameters);

    // Size up in 14px increments
    const size = props.frame
      ? AVATAR_SIZES[
          Math.min(
            Math.floor(Math.max(props.frame.width, props.frame.height) / 14),
            AVATAR_SIZES.length - 1,
          )
        ]
      : 'md';

    const sizePx = props.frame
      ? Math.min(props.frame.width, props.frame.height)
      : undefined;

    const widthHashtag = hashtags.find((hashtag) => hashtag.startsWith('w-'));
    const widthMatch = widthHashtag ? widthHashtag.match(/w-(\d+)/) : undefined;
    const width = widthMatch ? Number(widthMatch[1]) : undefined;
    const tailwindSize = width ? `${width * 0.25}rem` : undefined;

    return h(Avatar, {
      ...applyCommonProps(props),
      size,
      name,
      src,
      style: {
        aspectRatio: '1',
        ...(sizePx && {
          width: `${sizePx}px`,
          height: `${sizePx}px`,
        }),
        ...(tailwindSize && {
          width: tailwindSize,
          height: tailwindSize,
        }),
      },
      className: getBlockClassName(hashtags),
    });
  },
};
