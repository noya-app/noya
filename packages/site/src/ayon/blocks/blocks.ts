import Sketch from 'noya-file-format';
import {
  heroButtonRowSymbol,
  heroHeadlineStackSymbol,
  heroSymbol,
} from '../symbols/composed/HeroSymbol';
import { avatarSymbol } from '../symbols/primitive/AvatarSymbol';
import { boxSymbol } from '../symbols/primitive/BoxSymbol';
import { buttonSymbol } from '../symbols/primitive/ButtonSymbol';
import { checkboxSymbol } from '../symbols/primitive/CheckboxSymbol';
import { tagSymbol } from '../symbols/primitive/TagSymbol';
import { heroSymbolV1Id } from '../symbols/symbolIds';

export const allSymbols = [
  avatarSymbol,
  buttonSymbol,
  boxSymbol,
  heroSymbol,
  heroHeadlineStackSymbol,
  heroButtonRowSymbol,
  tagSymbol,
  checkboxSymbol,
];

export const allInsertableSymbols = allSymbols.filter(
  (symbol) => !symbol.blockDefinition?.isPassthrough,
);

export const symbolMap: Record<string, Sketch.SymbolMaster> = {
  ...Object.fromEntries(allSymbols.map((symbol) => [symbol.symbolID, symbol])),
  // Add extra mappings here (e.g. heroV1 => heroV2)
  [heroSymbolV1Id]: heroSymbol,
};
