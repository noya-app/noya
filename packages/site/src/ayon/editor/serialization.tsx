import Sketch from 'noya-file-format';
import { SketchModel } from 'noya-sketch-model';
import { BlockContent, Overrides } from 'noya-state';
import { Descendant, Node } from 'slate';
import { flattenPassthroughLayers } from '../blocks/flattenPassthroughLayers';
import { ParagraphElement } from './types';

export type EditorBlockContent = Omit<
  Required<BlockContent>,
  'normalizedText'
> & {
  layerPath: string[];
};

export function toContent(
  symbol: Sketch.SymbolMaster,
  nodes: Descendant[],
): EditorBlockContent {
  const layers = flattenPassthroughLayers(symbol);
  const overrides: Sketch.OverrideValue[] = [];

  for (let i = 0; i < layers.length; i++) {
    const layer = layers[i];
    const node = nodes[i + 1];
    const value = node ? Node.string(node) : '';

    overrides.push(
      SketchModel.overrideValue({
        overrideName: Overrides.encodeName([layer.do_objectID], 'blockText'),
        value,
      }),
    );
  }

  const rootNode = nodes[0] as ParagraphElement;

  return {
    symbolId: symbol.symbolID,
    blockText: nodes.length > layers.length ? Node.string(rootNode) : '',
    overrides: overrides,
    layerPath: [],
  };
}

export function extractBlockContent(
  instance: Sketch.SymbolInstance,
): EditorBlockContent {
  return {
    blockText: instance.blockText ?? '',
    overrides: instance.overrideValues,
    symbolId: instance.symbolID,
    layerPath: [],
  };
}

export function fromContent(
  symbol: Sketch.SymbolMaster,
  content: EditorBlockContent,
): Descendant[] {
  const layers = flattenPassthroughLayers(symbol);

  const layerNodes = layers.map((layer): ParagraphElement => {
    const value = Overrides.getOverrideValue(
      content.overrides ?? [],
      layer.do_objectID,
      'blockText',
    );

    return {
      type: 'paragraph',
      children: [{ text: value ?? '' }],
      symbolId: layer.symbolID,
      layerPath: [layer.do_objectID],
      isRoot: false,
    };
  });

  const rootNode: ParagraphElement = {
    type: 'paragraph',
    children: [{ text: content.blockText ?? '' }],
    symbolId: content.symbolId,
    layerPath: [],
    isRoot: true,
  };

  return [rootNode, ...layerNodes];
}

export function toString(value: Descendant[]): string {
  return value.map((n) => Node.string(n)).join('\n');
}
