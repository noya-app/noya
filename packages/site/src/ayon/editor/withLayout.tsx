import Sketch from 'noya-file-format';
import { Layers } from 'noya-state';
import { Node, Element as SlateElement, Transforms } from 'slate';
import { Blocks } from '../blocks/blocks';
import { flattenPassthroughLayers } from '../blocks/flattenPassthroughLayers';
import { CustomEditor, ParagraphElement } from './types';

export function insertBlock(
  editor: CustomEditor,
  layer: Sketch.SymbolInstance | Sketch.SymbolMaster,
  path: number[],
) {
  const block = Blocks[layer.symbolID];

  const paragraph: ParagraphElement = {
    type: 'paragraph',
    children: [{ text: '' }],
    label: block.symbol.name,
    symbolId: block.symbol.symbolID,
    placeholder: Layers.isSymbolInstance(layer) ? layer.blockText : undefined,
    layerId: layer ? layer.do_objectID : undefined,
  };

  Transforms.insertNodes(editor, paragraph, {
    at: path.concat(editor.children.length),
  });
}

export function setNestedNodeProperties(
  editor: CustomEditor,
  layer: Sketch.SymbolInstance,
  path: number[],
  node: ParagraphElement,
) {
  const block = Blocks[layer.symbolID];

  const label = block.symbol.name;
  const placeholder = layer ? layer.blockText : undefined;

  if (
    node.label === label &&
    node.placeholder === placeholder &&
    node.layerId === layer.do_objectID
  ) {
    return;
  }

  const newProperties: Partial<SlateElement> = {
    label,
    symbolId: layer.symbolID,
    placeholder,
    layerId: layer.do_objectID,
  };

  Transforms.setNodes<SlateElement>(editor, newProperties, {
    at: path,
  });
}

export function setRootNodeProperties(
  editor: CustomEditor,
  layer: Sketch.SymbolMaster,
  path: number[],
  node: ParagraphElement,
) {
  if (!node.label) return;

  const newProperties: Partial<SlateElement> = {
    label: undefined,
    symbolId: layer.symbolID,
    placeholder: undefined,
    layerId: undefined,
  };

  Transforms.setNodes<SlateElement>(editor, newProperties, {
    at: path,
  });
}

export function withLayout(initialSymbolId: string, editor: CustomEditor) {
  const { normalizeNode } = editor;

  editor.symbolId = initialSymbolId;

  editor.normalizeNode = ([node, path]) => {
    const symbol = Blocks[editor.symbolId].symbol;

    const layers = flattenPassthroughLayers(symbol);

    // If this is the editor
    if (path.length === 0) {
      // Ensure there's a node for each layer and the container
      while (editor.children.length < layers.length + 1) {
        insertBlock(editor, layers[editor.children.length - 1] ?? symbol, path);
      }

      while (editor.children.length > layers.length + 1) {
        Transforms.removeNodes(editor, {
          at: path.concat(editor.children.length - 1),
        });
      }

      for (const [child, childPath] of Node.children(editor, path)) {
        if (!SlateElement.isElement(child)) continue;

        const layer = layers[childPath[0]];

        if (layer) {
          setNestedNodeProperties(editor, layer, childPath, child);
        } else {
          setRootNodeProperties(editor, symbol, childPath, child);
        }
      }
    }

    return normalizeNode([node, path]);
  };

  return editor;
}
