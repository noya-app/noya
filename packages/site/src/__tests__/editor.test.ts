import { readFileSync } from 'fs';
import { SketchModel } from 'noya-sketch-model';
import { Layers, Overrides } from 'noya-state';
import path from 'path';
import { createEditor, Editor } from 'slate';
import { withHistory } from 'slate-history';
import { withReact } from 'slate-react';
import { Blocks } from '../ayon/blocks/blocks';
import { heading5SymbolId, heroSymbolV2Id } from '../ayon/blocks/symbolIds';
import { fromSymbol } from '../ayon/editor/serialization';
import { withLayout } from '../ayon/editor/withLayout';

// Jest doesn't know how to import a text file, so we mock it
jest.mock('../../safelist.txt', () => {
  return {
    default: readFileSync(path.join(__dirname, '../../safelist.txt'), 'utf8'),
  };
});

test('serializes empty block', () => {
  const layer = SketchModel.symbolInstance({
    symbolID: heroSymbolV2Id,
    frame: SketchModel.rect({
      width: 400,
      height: 400,
    }),
  });

  const editor = withLayout(
    layer.symbolID,
    withHistory(withReact(createEditor())),
  );

  Editor.normalize(editor, { force: true });

  const nodes = fromSymbol(Blocks[layer.symbolID].symbol, layer);

  expect(editor.children).toEqual(nodes);
});

test('serializes block with layer text', () => {
  const instance = SketchModel.symbolInstance({
    symbolID: heroSymbolV2Id,
    frame: SketchModel.rect({
      width: 400,
      height: 400,
    }),
  });

  Blocks[instance.symbolID].symbol.layers
    .filter(Layers.isSymbolInstance)
    .filter((layer) => layer.symbolID === heading5SymbolId)
    .forEach((layer) => {
      instance.overrideValues.push(
        SketchModel.overrideValue({
          overrideName: Overrides.encodeName([layer.do_objectID], 'blockText'),
          value: 'hello',
        }),
      );
    });

  const nodes = fromSymbol(Blocks[instance.symbolID].symbol, instance);

  expect(nodes).toMatchSnapshot();
});
