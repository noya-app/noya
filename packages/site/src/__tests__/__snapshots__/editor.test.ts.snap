// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes block with layer text 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "label": "Heading2",
    "layerId": "ef2d5b26-aa1c-40d3-8bab-37c10bccc5cb",
    "placeholder": "Create, iterate, inspire.",
    "symbolId": "b3884865-3fbe-4ed0-b577-c3f5864c4a21",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "hello",
      },
    ],
    "label": "Heading5",
    "layerId": "aa722c35-9ba4-4bf3-a5d0-f7d17f02c361",
    "placeholder": "Turn great ideas into new possibilities. #mb-4",
    "symbolId": "a6f2653e-1595-4dcc-b979-4e0b1340fda3",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "label": "Button",
    "layerId": "6b386c69-d6cf-4c2f-ae06-c92af43268d5",
    "placeholder": "Get started #primary #md",
    "symbolId": "cdf2346b-cb21-4f23-8d93-7c4fb2e3a5a0",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "label": "Link",
    "layerId": "eee85c94-7361-4bcf-8afb-f59c6e8661f7",
    "placeholder": "Learn more #text-slate-800 #icon-arrow-forward",
    "symbolId": "5be80b87-7bd6-4090-8947-316d54064544",
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "layerId": undefined,
    "placeholder": undefined,
    "symbolId": "61eb943b-7ee7-4b33-9315-aac7cb2d2764",
    "type": "paragraph",
  },
]
`;
